pkg/auth/providers/azure/clients/ad_graph_client.go

```diff
@@ -187,6 +187,7 @@ func NewAzureADGraphClientFromADALToken(config *v32.AzureADConfig, adalTokenSecr
 	if err != nil {
 		return nil, err
 	}
+	spt.SetRefreshCallbacks(nil)
```

Comment:
Would you mind elaborating on why this is necessary? From what I can see, you set the callbacks to nil when you construct the client, and I don't see any internal functionality that would result in this being set to not nil. Is this maybe an artifact of `callbacks ...TokenRefreshCallback` getting converted in a weird way?

Reply:
The construction of the client does not actually set `refreshCallbacks` to nil in line 145 and line 185. The constructors are variadic functions that take `TokenRefreshCallback` which is a function type. When you pass in nil in the constructors you are creating a list of one nil function.
``` golang
callbacks := []TokenRefreshCallback{nil}
```
instead, we want the list to be nil which is done by not providing any arguments. If you make that change you shouldn't need to call `SetRefreshCallbacks(nil)`

Reply:
@KevinJoiner, good call, thanks!